From 3fee779a6083ed39c31f1fb7d6486ab3921246d4 Mon Sep 17 00:00:00 2001
From: Wang Mingyu <wangmy@fujitsu.com>
Date: Tue, 27 Jun 2023 15:17:17 +0800
Subject: python3-hatchling: upgrade 1.17.0 -> 1.18.0

Changed:
--------
  Drop support for Python 3.7

Added:
--------
  Update the list of directories that are always excluded for builds

Fixed:
-------
  Fix dev mode when the project has symlinks and file inclusion is defined with the packages or only-include options
  Change the name of generated PTH files for dev mode so they come first lexicographically and therefore load first

Signed-off-by: Wang Mingyu <wangmy@fujitsu.com>
Signed-off-by: Richard Purdie <richard.purdie@linuxfoundation.org>
---
 .../recipes-devtools/python/python3-hatchling_1.18.0.bb | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)
 create mode 100644 meta/recipes-devtools/python/python3-hatchling_1.18.0.bb

(limited to 'meta/recipes-devtools/python/python3-hatchling_1.18.0.bb')

diff --git a/meta/recipes-devtools/python/python3-hatchling_1.18.0.bb b/meta/recipes-devtools/python/python3-hatchling_1.18.0.bb
new file mode 100644
index 0000000000..c94e49dab3
--- /dev/null
+++ b/meta/recipes-devtools/python/python3-hatchling_1.18.0.bb
@@ -0,0 +1,17 @@
+SUMMARY = "The extensible, standards compliant build backend used by Hatch"
+HOMEPAGE = "https://hatch.pypa.io/"
+LICENSE = "MIT"
+LIC_FILES_CHKSUM = "file://LICENSE.txt;md5=cbe2fd33fc9297692812fc94b7d27fd9"
+
+inherit pypi python_hatchling
+
+DEPENDS += "python3-pluggy-native python3-pathspec-native python3-packaging-native python3-editables-native python3-trove-classifiers-native"
+DEPENDS:remove:class-native = "python3-hatchling-native"
+
+SRC_URI[sha256sum] = "50e99c3110ce0afc3f7bdbadff1c71c17758e476731c27607940cfa6686489ca"
+
+do_compile:prepend() {
+    export PYTHONPATH=src
+}
+
+BBCLASSEXTEND = "native nativesdk"
-- 
cgit 1.2.3-korg

